Sources/Hyperspace/Service/Transport/TransportService.swift

```diff
@@ -42,19 +43,25 @@ extension TransportService: Transporting {
             self?.networkActivityController?.stop()
             self?.handle(data: data, response: response, error: error, for: HTTP.Request(urlRequest: request), completion: completion)
         }
-        
-        tasks[request] = task
-        task.resume()
-        
-        networkActivityController?.start()
+
+        queue.async { [weak self] in
+            self?.tasks[request] = task
+            task.resume()
+
+            self?.networkActivityController?.start()
+        }
     }
     
     public func cancelTask(for request: URLRequest) {
-        tasks[request]?.cancel()
+        queue.async { [weak self] in
```

Comment:
I feel like this should break some tests no? I think, because this is a `private` queue we're using, and we're careful how we use it we can safely `queue.sync` as well, which might fix any of the broken assumptions in the tests.

Reply:
Hah, this gave me some pause, but ultimately I do think we can make the call synchronous.